app/src/main/java/app/revanced/manager/ui/viewmodel/PatcherViewModel.kt

```diff
@@ -238,23 +283,67 @@ class PatcherViewModel(
         context.startActivity(shareIntent)
     }
 
+    private fun versionNameToInt(versionName: String): Int {
+        val versionParts = versionName.split(".")
+        return versionParts[0].toInt() * 10000 + versionParts[1].toInt() * 100 + versionParts[2].toInt()
+    }
+
     fun open() = installedPackageName?.let(pm::launch)
 
     fun install(installType: InstallType) = viewModelScope.launch {
         try {
             isInstalling = true
+
+            // If the app is currently installed
+            val existingPackageInfo = pm.getPackageInfo(packageName)
+            if (existingPackageInfo != null) {
+                // Check if the app version is less than the installed version
+                if (versionNameToInt(input.selectedApp.version) < versionNameToInt(existingPackageInfo.versionName)) {
+                    // Exit if the selected app version is less than the installed version
+                    installerStatusDialogModel.packageInstallerStatus = PackageInstaller.STATUS_FAILURE_CONFLICT
+                    return@launch
+                }
+            }
+
+            val currentPackageInfo = pm.getPackageInfo(outputFile)
+                ?: throw Exception("Failed to load application info")
+
             when (installType) {
                 InstallType.DEFAULT -> {
+                    // Check if the app is mounted as root
+                    // If it is, unmount it first, silently
+                    if (rootInstaller.hasRootAccess() && rootInstaller.isAppMounted(packageName)) {
+                        rootInstaller.unmount(packageName)
+                    }
+
+                    // Install regularly
                     pm.installApp(listOf(outputFile))
                 }
 
                 InstallType.ROOT -> {
                     try {
+                        // Check for base APK, first check if the app is already installed
+                        if (existingPackageInfo == null) {
+                            val inputPackageInfo = if (inputFile != null)
+                                pm.getPackageInfo(inputFile!!)
+                            else
+                                null
```

Comment:
Use `?.let` instead. `val inputPackageInfo = inputFile?.let { pm.getPackageInfo(it) }`